tools/detect_targets.py

```diff
@@ -68,14 +70,17 @@ def main():
         # parameters like 'toolchains_filter' are also set.
         muts = get_autodetected_MUTS_list()
 
+        mcu_filter = options.general_filter_regex or [".*"]
```

Comment:
Nit: Any reason why the default value for the `--filter` argument can't be `".*"`?

It seems kind of odd to accept an array of regex expressions. Wouldn't you just change your regex to be more accommodating?

Also, the current ArgParse option will set `options.general_filter_regex` to just a string, not an array. Won't this blow up below?